File: CarePackage/Software/DotNet8Runtime.cs
namespace CarePackage.Software;

public class DotNet8Runtime : BaseSoftware
{
    public override string           Key            { get; set; } = "dotnet-8-runtime";
    public override string           Name           { get; set; } = ".NET 8.0 Runtime";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Runtime;
    public override string           DownloadName   { get; set; } = "dotnet-runtime-8.0-win-x64.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.dotnet;
    public override string           Homepage       { get; set; } = "https://dot.net";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://download.visualstudio.microsoft.com/download/pr/53e9e41c-b362-4598-9985-45f989518016/53c5e1919ba2fe23273f2abaff65595b/dotnet-runtime-8.0.11-win-x64.exe");
}

File: src/lib/software/dotnet_8_runtime.py
from src.lib.software import BaseSoftware, SoftwareCategory

class DotNet8Runtime(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'dotnet-8-runtime'
        self.name = '.NET 8.0 Runtime'
        self.category = SoftwareCategory.DotNet
        self.download_name = 'dotnet-runtime-8.0-win-x64.exe'
        self.is_archive = False
        self.should_cache_url = False
        self.requires_admin = False
        self.icon = 'dotnet.png'
        self.homepage = 'https://dot.net'

    def resolve_download_url(self):
        self.url_resolved.emit('https://download.visualstudio.microsoft.com/download/pr/53e9e41c-b362-4598-9985-45f989518016/53c5e1919ba2fe23273f2abaff65595b/dotnet-runtime-8.0.11-win-x64.exe')
